Test failures

```
[ERROR] Errors: 
[ERROR]   ITestS3AFailureHandling.testMultiObjectDeleteLargeNumKeys » OutOfMemory Direct...
[ERROR]   ITestS3ADeleteFilesOneByOne.testBulkRenameAndDelete » OutOfMemory Direct buffe...
[ERROR]   ITestS3SelectLandsat.testSelectSeekFullLandsat:427->AbstractS3SelectTest.seek:701 » TestTimedOut
[INFO] 
```

* OOM in HADOOP-17937; ITestS3ADeleteFilesOneByOne. testBulkRenameAndDelete. This shows we would benefit from lazy creation of buffers.
* ITestS3AStorageClass.testCreateAndCopyObjectWithStorageClassGlacier
 
```
[ERROR] Failures: 
[ERROR]   ITestS3AStorageClass.testCreateAndCopyObjectWithStorageClassGlacier:129->assertObjectHasStorageClass:215 [Storage class of object s3a://stevel-london/fork-0001/test/testCreateAndCopyObjectWithStorageClassGlacier/file1] 
Expecting:
 <null>
to be equal to:
 <"glacier">
ignoring case considerations
[ERROR]   ITestS3AStorageClass.testCreateAndCopyObjectWithStorageClassReducedRedundancy:104->assertObjectHasStorageClass:215 [Storage class of object s3a://stevel-london/fork-0001/test/testCreateAndCopyObjectWithStorageClassReducedRedundancy/file1] 
Expecting:
 <null>
to be equal to:
 <"reduced_redundancy">
ignoring case considerations
```


I will need to see if this is a regression. I believe i have disabled setting storage class.

```
* [ERROR] testSelectSeekFullLandsat(org.apache.hadoop.fs.s3a.select.ITestS3SelectLandsat)  Time elapsed: 600.013 s  <<< ERROR!
org.junit.runners.model.TestTimedOutException: test timed out after 600000 milliseconds
```

--- a/hadoop-project/pom.xml
+++ b/hadoop-project/pom.xml
@@ -184,7 +184,7 @@
     <exec-maven-plugin.version>1.3.1</exec-maven-plugin.version>
     <make-maven-plugin.version>1.0-beta-1</make-maven-plugin.version>
     <surefire.fork.timeout>900</surefire.fork.timeout>
-    <aws-java-sdk.version>1.12.132</aws-java-sdk.version>
+    <aws-java-sdk.version>1.12.262</aws-java-sdk.version>
     <hsqldb.version>2.3.4</hsqldb.version>
     <frontend-maven-plugin.version>1.11.2</frontend-maven-plugin.version>
     <jasmine-maven-plugin.version>2.1</jasmine-maven-plugin.version>

--- a/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/testing.md
+++ b/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/testing.md
@@ -1291,7 +1291,7 @@ time bin/hadoop fs -copyFromLocal -t 10  share/hadoop/tools/lib/*aws*jar $BUCKET
 # expect the iostatistics object_list_request value to be O(directories)
 bin/hadoop fs -ls -R $BUCKET/
 
-# expect the iostatistics object_list_request and op_get_content_summary values to be 1 
+# expect the iostatistics object_list_request and op_get_content_summary values to be 1
 bin/hadoop fs -du -h -s $BUCKET/
 
 mkdir tmp

--- a/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/testing.md
+++ b/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/testing.md
@@ -1156,6 +1156,14 @@ We need a run through of the CLI to see if there have been changes there
 which cause problems, especially whether new log messages have surfaced,
 or whether some packaging change breaks that CLI.
 
+It is always interesting when doing this to enable IOStatistics reporting
+```xml
+<property>
+  <name>fs.iostatistics.logging.level</name>
+  <value>info</value>
+</property>
+```
+
 From the root of the project, create a command line release `mvn package -Pdist -DskipTests -Dmaven.javadoc.skip=true  -DskipShade`;
 
 1. Change into the `hadoop-dist/target/hadoop-x.y.z-SNAPSHOT` dir.
@@ -1275,6 +1283,21 @@ bin/hadoop s3guard markers -clean -verbose $BUCKET
 bin/hadoop s3guard markers -audit -verbose $BUCKET
 
 # ---------------------------------------------------
+# Copy to from local
+# ---------------------------------------------------
+
+time bin/hadoop fs -copyFromLocal -t 10  share/hadoop/tools/lib/*aws*jar $BUCKET/
+
+# expect the iostatistics object_list_request value to be O(directories)
+bin/hadoop fs -ls -R $BUCKET/
+
+# expect the iostatistics object_list_request and op_get_content_summary values to be 1 
+bin/hadoop fs -du -h -s $BUCKET/
+
+mkdir tmp
+time bin/hadoop fs -copyToLocal -t 10  $BUCKET/\*aws\* tmp
+
+# ---------------------------------------------------
 # S3 Select on Landsat
 # ---------------------------------------------------
 
@@ -1282,6 +1305,17 @@ export LANDSATGZ=s3a://landsat-pds/scene_list.gz
 
 bin/hadoop s3guard select -header use -compression gzip $LANDSATGZ \
  "SELECT s.entityId,s.cloudCover FROM S3OBJECT s WHERE s.cloudCover < '0.0' LIMIT 100"
+
+
+# ---------------------------------------------------
+# Cloudstore
+# check out and build https://github.com/steveloughran/cloudstore
+# then for these tests, set CLOUDSTORE env var to point to the JAR
+# ---------------------------------------------------
+
+bin/hadoop jar $CLOUDSTORE storediag $BUCKET
+
+time bin/hadoop jar $CLOUDSTORE bandwidth 64M $BUCKET/testfile
 
 ```
 

